lighthouse-core/test/results/sample_v2.json

```diff
@@ -7020,6 +7020,12 @@
         "duration": 100,
         "entryType": "measure"
       },
+      {
+        "startTime": 0,
+        "name": "lh:computed:JSBundles",
```

Comment:
I think we are recomputing because `artifacts` can contain more than `SourceMaps` and `ScriptElements` when we pass it to `JsBundles.request`.

Reply:
Good catch. JSBundles isn't the only computed artifact that suffers from this, so I'll put up a PR fixing every case after this.

should we: 1) explicitly pass the keys needed or 2) introduce a second param to `makeComputedArtifact` that will do some preprocessing of the input params given to it 3) do 1. but add a helper `pick` fn

Reply:
> 2) introduce a second param to makeComputedArtifact that will do some preprocessing of the input params given to it

Slight preference for this because I like passing `artifacts` in without much fuss. Also seems easier than validating every usage of every computed artifact.

Second param could be a shell object whose keys are the same as `artifacts`. `request` would only check equality on the keys of the shell object passed in. Not super pretty but it would get TS to validate everything:

```js
module.exports = makeComputedArtifact(JSBundles, {
  'ScriptElements': undefined,
  'SourceMaps': undefined,
});
```

Reply:
https://stackoverflow.com/questions/60131681/make-sure-array-has-all-types-from-a-union

Interesting, might be a way to do this with an array:

```js
module.exports = makeComputedArtifact(JSBundles, ['ScriptElements', 'SourceMaps']);
```